### DISCUSSION TITLE
Self-hosting Supabase

### DISCUSSION THREAD

**Discussion by dokgu** [supabase/supabase#15704] [Questions]
Hi guys, I need to self-host Supabase on a Google Cloud VM but I'm having issues setting it up (at least based on what I see locally). I tried to install it locally on my machine just to test it out first before starting a new VM on Google Cloud infrastructure.

After spinning up the Docker container, I visited localhost:3000 and somehow I didn't even get to a login page - I was straight into the dashboard right away and that doesn't feel secure to me. The pages are supposed to be protected and only available once I log in but I couldn't figure out how to do that. Is there a configuration that I'm missing?

Another issue I have is that the self-hosted solution only allows for one organization and one project - I'm fine with being limited to one organization but I do need to be able to create multiple projects under that organization and have different keys for each project. Is this possible at all? For those who were able to set that up and get it to work, any guides you followed or any documentation that is available?

**Comment by gyanprabhat7** [comment]
Hey did you find any solution for it?

**Reply by dokgu** [reply]
Not really, you can use tools like Authelia / Keycloak / Authentik to put your self-hosted instance of Supabase behind their authentication but if I remember correctly they've already added a pretty basic authentication method the last time I tried to spin up a new instance.

In terms of having multiple projects, I am not sure because I haven't been following Supabase that much - I'm currently using Appwrite for my projects right now.
